src/commands/compare-command.js

```diff
@@ -0,0 +1,79 @@
+import { Client, MessageEmbed } from 'discord.js';
+import { UserDB } from './../components/user-db';
+import { api, platforms } from 'call-of-duty-api-es6';
+import _ from 'lodash';
+
+export class CompareCommand {
+    static getCommand() {
+        return 'compare';
+    }
+
+    static create(message, logger, client) {
+        return new CompareCommand(message, logger, client);
+    }
+
+    constructor(message, logger, client) {
+        this.message = message;
+        this.logger = logger;
+        this.client = client;
+        this.API = new api();
+    }
+
+    async execute() {
+        try {
+            let ids = [];
+            for (const [key, value] of (this.client.users.cache.entries())) {
+                if (!value.bot) {
+                    ids.push(key);
+                }
+            }
+            if (!ids.length) {
+
+            }
+            const userDb = new UserDB();
+            let docs = await userDb.getUsers(ids);
+            let stats = await Promise.all(docs.map(doc => this.API.MWstats(doc.username, doc.platform)));
```

Comment:
tried my best to combine all these, any suggestions here?

Reply:
I think you can do:
```
let stats = docs.map(async doc => await this.API.MWstats(doc.username, doc.platform)));
```
docs is not a promise and only the MWstats returns a promise which we wait for in the lambda.